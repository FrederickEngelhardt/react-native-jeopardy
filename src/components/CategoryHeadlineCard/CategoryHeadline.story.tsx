import * as React from "react";
import { storiesOf } from "@storybook/react-native";
import {
  withKnobs
} from "@storybook/addon-knobs";
import {action} from "@storybook/addon-actions";

import CategoryHeadlineCard from "./CategoryHeadlineCard";
import categoryHeadlineProps from "./CategoryHeadlineCard.props";


storiesOf("<CategoryHeadlineCard />", module)
  .addDecorator(withKnobs)
  .add("default", () => (
    // @ts-ignore
    <CategoryHeadlineCard
      {...categoryHeadlineProps}
      updateUserScore={action('updated user score')}
    />
  ));
