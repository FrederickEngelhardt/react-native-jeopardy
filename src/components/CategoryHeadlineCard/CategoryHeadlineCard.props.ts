const categoryColumnProps = {
	title: 'Characters'
}

export default categoryColumnProps
