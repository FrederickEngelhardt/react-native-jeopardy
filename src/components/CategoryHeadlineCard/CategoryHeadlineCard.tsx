import React from "react";
import styled from "styled-components";
import { CardContainer, TitleText } from "../Card/Card";

export interface Props {
  title: string;
}

const HeadlineText = styled(TitleText)`
  font-weight: 900;
  font-size: 48px;
`;

const CategoryHeadlineCard = ({ title }: Props) => {
  return (
    <CardContainer>
      <HeadlineText>{title}</HeadlineText>
    </CardContainer>
  );
};

export default CategoryHeadlineCard;
