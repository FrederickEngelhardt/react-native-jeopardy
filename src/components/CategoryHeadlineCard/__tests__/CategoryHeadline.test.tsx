import React from "react";
import { render } from "@testing-library/react-native";

import CategoryHeadlineCard from "../CategoryHeadlineCard";
import categoryHeadlineProps from "../CategoryHeadlineCard.props";

describe("<CategoryHeadline/>", () => {
  it("renders correctly", () => {
    const { container } = render(
        <CategoryHeadlineCard {...categoryHeadlineProps} />
    );
    expect(container).toMatchSnapshot();
  });
});
