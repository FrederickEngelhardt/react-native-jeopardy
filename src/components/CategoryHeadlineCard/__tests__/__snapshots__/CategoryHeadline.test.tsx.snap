// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryHeadline/> renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#d1d1d1",
          "marginBottom": 12,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 48,
            "fontWeight": "900",
            "textAlign": "center",
          },
        ]
      }
    >
      Characters
    </Text>
  </View>
</View>
`;
